spec/datadog/core/remote/component_spec.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'datadog/core/remote/component'
+
+RSpec.describe Datadog::Core::Remote::Component do
+  let(:settings) { Datadog::Core::Configuration::Settings.new }
+  let(:agent_settings) { Datadog::Core::Configuration::AgentSettingsResolver.call(settings, logger: nil) }
+
+  describe '#initialize' do
+    subject(:component) { described_class.new(settings, agent_settings) }
+
+    let(:transport_v7) { double }
+    let(:client) { double }
+    let(:worker) { component.instance_eval { @worker } }
+
+    before do
+      expect(Datadog::Core::Transport::HTTP).to receive(:v7).and_return(transport_v7)
+      expect(Datadog::Core::Remote::Client).to receive(:new).and_return(client)
+
+      expect(worker).to receive(:start).and_call_original
+      expect(worker).to receive(:stop).and_call_original
+
+      component.barrier(:once)
+    end
+
+    after do
+      component.shutdown!
+    end
+
+    context 'when client sync succeeds' do
+      before do
+        expect(worker).to receive(:call).and_call_original
+        expect(client).to receive(:sync).and_return(nil)
+      end
+
+      it 'does not log any error' do
+        expect(Datadog.logger).to_not receive(:error)
+
+        Thread.pass # allow worker thread to work
+        sleep 0.1
+      end
+    end
+
+    context 'when client sync raises' do
+      let(:second_client) { double }
+
+      before do
+        expect(worker).to receive(:call).and_call_original
+        expect(client).to receive(:sync).and_raise(StandardError, 'test')
+      end
+
+      it 'logs an error' do
+        allow(Datadog::Core::Remote::Client).to receive(:new).and_return(client)
+
+        expect(Datadog.logger).to receive(:error).and_return(nil)
+
+        Thread.pass # allow worker thread to work
+        sleep 0.1
+      end
```

Comment:
Doing expectations on `Datadog.logger` is not very valuable, in my opinion. They cause the test suite to be more brittle since if we remove the logline, we have to remove the test. I will focus on testing the actual behaviours. Ex:  `creates a new client`

Reply:
The goal of this test is precisely to test that an error-level log is emitted on exception catching.

Reply:
I still think that testing this kind of behaviour doesn't yield much benefits